packages/kbn-journeys/services/page/kibana_page.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { subj } from '@kbn/test-subj-selector';
+import { ToolingLog } from '@kbn/tooling-log';
+import { Page } from 'playwright';
+
+interface WaitForRenderArgs {
+  expectedItemsCount: number;
+  itemLocator: string;
+  checkAttribute: string;
+}
+
+export class KibanaPage {
+  readonly page: Page;
+  readonly log: ToolingLog;
+
+  constructor(page: Page, log: ToolingLog) {
+    this.page = page;
+    this.log = log;
+  }
+
+  async waitForHeader() {
+    return this.page.waitForSelector('.headerGlobalNav', {
+      state: 'attached',
+    });
+  }
+
+  async backToDashboardListing() {
+    await this.page.click(subj('breadcrumb dashboardListingBreadcrumb first'));
+  }
+
+  async waitForRender({ expectedItemsCount, itemLocator, checkAttribute }: WaitForRenderArgs) {
+    try {
+      await this.page.waitForFunction(
+        function renderCompleted(args: WaitForRenderArgs) {
+          const renderingItems = Array.from(document.querySelectorAll(args.itemLocator));
+          const allItemsLoaded = renderingItems.length === args.expectedItemsCount;
+          return allItemsLoaded
+            ? renderingItems.every((e) => e.getAttribute(args.checkAttribute) === 'true')
+            : false;
+        },
+        { expectedItemsCount, itemLocator, checkAttribute }
+      );
+    } catch (err) {
+      const loaded = await this.page.$$(itemLocator);
+      const rendered = await this.page.$$(`${itemLocator}[${checkAttribute}="true"]`);
+      this.log.error(
+        `'waitForRendering' failed: loaded - ${loaded.length}, rendered - ${rendered.length}, expected count - ${expectedItemsCount}`
+      );
+      throw err;
+    }
+  }
+
+  async waitForVisualizations(count: number) {
+    await this.waitForRender({
+      expectedItemsCount: count,
+      itemLocator: '[data-rendering-count]',
+      checkAttribute: 'data-render-complete',
```

Comment:
Are these in a typed enumeration somewhere?  Is there a possibility that these values can be changed "out from under" this code?  Could be just a nit, but it'd be nice to get some strongly typed support on these strings.

If not an enum, something like the following could be an option:
```ts
const animals = ["cat", "dog", "mouse"] as const;
type Animal = (typeof animals)[number];
const someA: Animal = "cat";
console.log(`\nλjs someA: \n  ${someA}`)

const never: Animal = "horse"
```

Reply:
```
interface WaitForRenderArgs {
  expectedItemsCount: number;
  itemLocator: string;
  checkAttribute: string;
}
```

https://github.com/elastic/kibana/pull/162902/files#diff-5b60988a242d963599779ad6ddd066dc2ad26266a68e5860433fe3828c934673R13-R17

Reply:
Indeed, but I'm referring to the string literal values changing out from under us.
But, if that's not a problem then no worries.

Reply:
Again, could be seen as just a nit.

Reply:
I see your point. The values are hardcoded in tsx files defining component structure in DOM. I also hardcode them as is and assume that if someone will change it in component they will update all occurrence in Kibana repo.

I don't think that there is much value to strongly type playwright locators here, but would keep the idea for some other cases.